lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -882,13 +882,10 @@ void HdVP2Mesh::Sync(
         || HdChangeTracker::IsPrimvarDirty(*dirtyBits, id, HdTokens->primvar) || instancerDirty) {
 
         auto addRequiredPrimvars = [&](const SdfPath& materialId) {
-            const HdVP2Material* material = static_cast<const HdVP2Material*>(
-                renderIndex.GetSprim(HdPrimTypeTokens->material, materialId));
-            TfToken              materialNetworkToken = _GetMaterialNetworkToken(reprToken);
-            const TfTokenVector& requiredPrimvars
-                = material && material->GetSurfaceShader(materialNetworkToken)
-                ? material->GetRequiredPrimvars(materialNetworkToken)
-                : sFallbackShaderPrimvars;
+            TfTokenVector requiredPrimvars;
+            if (!_GetMaterialPrimvars(renderIndex, materialId, requiredPrimvars)) {
+                requiredPrimvars = sFallbackShaderPrimvars;
+            }
```

Comment:
Since primvars (and their vertex buffers) are shared among all representations, we need to include here all required primvars from all available material networks, which is done inside _GetMaterialPrimvars method.